modules/dists/CyclicDist.chpl

```diff
@@ -646,12 +646,15 @@ iter CyclicDom.these(param tag: iterKind) where tag == iterKind.leader {
       // distribution (at least, I couldn't figure out a way to not go
       // back and forth without breaking tests)
       const zeroShift = {(...newFollowThis)}.chpl__unTranslate(wholeLow);
-      var result: rank*range(idxType=idxType, stridable=true);
+      var result: rank*range(idxType=idxType, strides=chpl_strideProduct(
+                              whole.strides, chpl_strideUnion(zeroShift) ));
       type strType = chpl__signedType(idxType);
       for param i in 0..rank-1 {
         const wholestride = chpl__tuplify(wholeStride)(i);
         const ref dim = zeroShift.dim(i);
-        result(i) = (dim.first / wholestride:idxType)..(dim.last / wholestride:idxType) by (dim.stride:strType / wholestride);
+        result(i).chpl_setFields(dim.first / wholestride:idxType,
```

Comment:
Is this ``chpl_setFields`` call necessary? Ideally we would write this kind of code in the same way that we would recommend for users.